src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryAddress.java

```diff
@@ -41,8 +41,8 @@
  * <p>
  * All implementations of this interface must be <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>;
  * programmers should treat instances that are {@linkplain #equals(Object) equal} as interchangeable and should not
- * use instances for synchronization, or unpredictable behavior may occur. The {@code equals} method should
- * be used for comparisons.
+ * use instances for synchronization, or unpredictable behavior may occur. For example, in a future release, synchronization may fail.
```

Comment:
Add a line break before the new sentence and in MemoryLayout.java and MemorySegment.java